src/Cello/problem_Problem.hpp

```diff
@@ -137,6 +137,14 @@ class Problem : public PUP::able
   /// Return the named method object if present
   Method * method (std::string name) const throw();
 
+  // Return whether a method object with given name exists for this problem
+  bool method_exists (std::string name) const throw();
+
+  // Returns true if method objects with both given names exist for this problem,
+  // and the method called "name1" precedes the method called "name2" in the
+  // method list. Returns false otherwise.
```

Comment:
You should specify the behavior in edge-case where you have more than one instances of the Method called  "name1" and the method called "name2" occurs between them. I'm pretty sure the function returns `true`.

This edge-case is not pathological, there are certain methods (like the flux correction) that are designed to support multiple executions per block per cycle.

Reply:
Thanks for pointing this out. I hadn't considered that methods might be executed more than once per cycle. 

I suppose what I really want to check is that "name2" does not precede any occurrence of "name1". I suppose then I should probably rename the function to `method_does_not_precede`, and then switch around the arguments in the calls to this function, but that would make it a bit more verbose. I think the intended meaning of `method_precede` is still clear enough.